src/Credentials/CredentialTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Aws\Credentials;
+
+use Aws\Exception\CredentialsException;
+use GuzzleHttp\Promise;
+
+/**
+ * @internal Sourcing credential profile data from file helpers.
+ */
+trait CredentialTrait
+{
+    /**
+     * Check profile availability in a given file.
+     * Returns a string describing any error that could occur.
+     *
+     * @param string $profile
+     * @param string $filename
+     *
+     * @return null|string
+     */
+    private static function checkProfile($profile, $filename)
+    {
+        if (!is_readable($filename)) {
+            return "Cannot read credentials from $filename";
+        }
+        $data = parse_ini_file($filename, true);
```

Comment:
Why are you not returning the loaded `$data`? Right now, you are running this method twice, loading and parsing the same INI file. This method should return the parsed ini file and we should not re-parse the file again.